chain/rosetta-rpc/src/adapters/mod.rs

```diff
@@ -119,6 +119,10 @@ pub(crate) async fn convert_block_to_transactions(
         .await?
         .unwrap();
 
+    // TODO(mina86): Do we actually need ‘seen’?  I’m kinda confused at this
+    // point how changes are stored in the database and whether view_client can
+    // return duplicate AccountTouched entries.
```

Comment:
In a single block a single account can be touched in several receipts / transactions and also receive rewards.

Well, to be honest, I feel it will be much easier to introduce a new view-client method to just return full state changes in a given block, so there is no need to fetch those through "give me all the affected accounts" -> "give me full state changes for all the affected accounts, and here is the list", and instead there is just "give me all state changes in the block".

Reply:
Yes, that’s exactly what I was thinking as well. I think it should be somewhat trivial to add `StateChangesRequestView::AllAccountsChanges` which would do exactly what we need.

For now I’m going to merge this so that the issue is resolved but I do plan to make the optimisation.